### PR TITLE
chore: fix complete script

### PR DESCRIPTION
**Pull request by ZeaLoVe**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. add auto-cleanup support
2. fix the wrong way of getting error_code, get code from error msg.

Fixes #issue

### CODE REVIEW

File: tests/logictest/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    create_name = statement_words[create_name_index]
+    need_cleanup_set.add((create_type, create_name.split('(')[0]))
+
+# if __name__ == '__main__':
+#     pick_create_statement("Create database db2;")
+#     pick_create_statement("drop table db1")
+#     pick_create_statement("CREATE database DB8;")
+#     pick_create_statement("create TABLE t2;")
+#     pick_create_statement("create user t2;")
+#     print(get_cleanup_statements())
```
Comment (leiysky) [inline tests/logictest/cleanup.py:52]: Shall we remove this?

Comment (ZeaLoVe) [reply tests/logictest/cleanup.py:52]: > Shall we remove this?

Yes, but this is my debug way without using pytest.

File: tests/logictest/cleanup.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    else:
+        create_type_index = 2
+    
+    if 'transient' in statement_lower:
+        create_type_index += 1
+        create_name_index += 1
+
+    create_type = statement_words[1] 
+    create_name = statement_words[create_name_index]
+    need_cleanup_set.add((create_type, create_name.split('(')[0]))
```
Comment (leiysky) [inline tests/logictest/cleanup.py:44]: IMO, it's better to implement this with regular expression.

Comment (ZeaLoVe) [reply tests/logictest/cleanup.py:44]: > IMO, it's better to implement this with regular expression.

I think just fine now, it meets current needs, maybe refactor it for further needs?

**Comment by drmingdrmer** [comment]
`main` branch is force pushed(**again** `:(`, to cleanup big object in history), update your working branch please: [To everybody: main branch is force pushed. #7263](https://github.com/datafuselabs/databend/issues/7263)

**Review by leiysky — APPROVED** [review]

**Review by xudong963 — APPROVED** [review]

File: tests/logictest/README.md [hunk 3]
```diff
@@ -161,7 +161,7 @@ select 1;
 1
 ```
 
-3. if some test has flaky failure, and you want ignore it, simply add skipped before statement query.(Remove it after problem solved)
+3. if some test has a flaky failure, and you want to ignore it, simply add skipped before statement query. (Remove it after the problem is solved)
 ```
 statement query skipped I
```
Comment (sundy-li) [inline tests/logictest/README.md:166]: if this is skipped, I think the `I` is useless.

complete script can't generate `I`.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**